src/sdk/Transcription/ConversationTranslator.ts

```diff
@@ -280,7 +280,7 @@ export class ConversationTranslator implements IConversationTranslator, IDisposa
                     ((error: any) => {
                         this.privIsSpeaking = false;
                         // this.fireCancelEvent(error);
-                        this.cancelSpeech();
+                        this.cancelSpeech().catch();
```

Comment:
>.catch(); [](start = 43, length = 9)

Review

Reply:
Ideally if we wanted to handle errors fully and properly here, we would append the errors from the cancelSpeech() operation (line 283) to the error returned in the call to handleError() at line 285 as an "inner error". Alternatively we may want to simply log the inner error and and not report it to the user.

Either way I think not waiting for the cancelSpeech() async operation to complete before invoking the callback may be problematic as we may return to the user in some transient internal state.

---
In reply to: [449317811](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/198#discussion_r449317811) [](ancestors = 449317811)